exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -513,56 +514,97 @@ object SchemaUtils {
      */
     fun checkMappingConsistence(vararg tables: Table, withLogs: Boolean = true): List<String> {
         if (withLogs) {
-            checkExcessiveIndices(tables = tables)
+            checkExcessiveForeignKeyConstraints(tables = tables, withLogs = true)
```

Comment:
I would drop `withLogs` flag and rely on the log level

Reply:
For computations, we can make our own inline function with `message: () -> String` argument or check if the logging library has one. 

@bog-walk, what do you think?

Reply:
I agree that I would drop `withLogs`, at least for the new generate function. 

We could check for the enabled level, but, for example, the results in [logTimeSpent()](https://github.com/JetBrains/Exposed/blob/main/exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt#L11) are meant for the INFO level, which I believe a lot of users set as the default. So relying on a guard like `exposedLogger.isInfoEnabled`, or the implicit level check in `info()`, might evaluate to `true` for many and result in users getting logs they don't want. Maybe if we were logging everything to DEBUG it would be easier.

Reply:
If you put an expression in a string template, it will be computed regardless of the log level:
```kotlin
logger.info("foo: ${heavyFunctionCall()}")
```

Otherwise, we could have an extension function with block:
```kotlin
logger.info { "foo: ${heavyFunctionCall()}" }
```
In this case, the string template is computed only when we call lambda (and we can condition it to the log level)

I think it's not critical for this PR, feel free to implement it later or ignore

Reply:
I'm going to leave this as is right now just to keep this PR small, but I made a note of it for the next PR.

Reply:
@e5l @joc-a If the end goal is to avoid these computations, we could also switch to parameterized logging, which is supported by slf4j. This would mean using the existing function variants that take a second argument:
```kt
exposedLogger.info("foo: {}", heavyFunctionCall())
```
According to the docs, if I'm understanding correctly:
> This form avoids superfluous object creation when the logger is disabled for the INFO level.